libc/string_tokpbrkstr.c

```diff
@@ -0,0 +1,633 @@
+	 */
+	TEST_ASSERT_EQUAL_PTR(extAsciiStr, strtok(extAsciiStr, "\128"));
+	for (i = ASCII_LENGTH + 1; i < sizeof(separator); i++) {
+		separator[i - ASCII_LENGTH] = i;
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i - ASCII_LENGTH], strtok(NULL, &separator[i - ASCII_LENGTH]));
+	}
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_tok_r)
+{
+}
+
+
+TEST_TEAR_DOWN(string_tok_r)
+{
+}
+
+
+TEST(string_tok_r, basic)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	/*
+	 * There needed to be two copies of the same element as strtok_r works
+	 * on the same space where the variable is settled when even using restrict
+	 */
+	char loremIpsum1[BUFFSIZE] = LOREM_IPSUM,
+		 loremIpsum2[BUFFSIZE] = LOREM_IPSUM,
+		 dividers[BUFFSIZE],
+		 asciiSupp[ASCII_LENGTH],
+		 *restState1 = NULL,
+		 *restState2 = NULL;
+
+	create_ascii_set(1);
+
+	sprintf(asciiSupp, "%s", asciiStr);
+
+
+	/* Assign elements from an array to restrict for further use */
+	TEST_ASSERT_EQUAL_PTR(loremIpsum1, strtok_r(loremIpsum1, " ", &restState1));
+	TEST_ASSERT_EQUAL_PTR(loremIpsum2, strtok_r(loremIpsum2, " ", &restState2));
+
+	/* searching for the next elements after breaking points in two arrays without ending previous tokenization*/
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum1[6], strtok_r(NULL, " ", &restState1));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum2[6], strtok_r(NULL, " ", &restState2));
+
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum1[12], strtok_r(NULL, " ", &restState1));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum2[12], strtok_r(NULL, " ", &restState2));
+
+	/* Checking if both of restricted strtok ended reading array and return NULL */
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, " ", &restState1));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, " ", &restState2));
+
+	sprintf(dividers, "%c%c%c", asciiStr[1], asciiStr[sizeof(asciiStr) / 2], asciiStr[sizeof(asciiStr) - 1]);
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strtok_r(asciiStr, dividers, &restState1));
+	TEST_ASSERT_EQUAL_PTR(asciiSupp, strtok_r(asciiSupp, dividers, &restState2));
+
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[2], strtok_r(NULL, dividers, &restState1));
+	TEST_ASSERT_EQUAL_PTR(&asciiSupp[2], strtok_r(NULL, dividers, &restState2));
+
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[sizeof(asciiStr) / 2 + 1], strtok_r(NULL, dividers, &restState1));
+	TEST_ASSERT_EQUAL_PTR(&asciiSupp[sizeof(asciiSupp) / 2 + 1], strtok_r(NULL, dividers, &restState2));
+
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, dividers, &restState1));
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, dividers, &restState2));
+}
+
+
+TEST(string_tok_r, null_char_impact)
+{
+#ifdef __phoenix__
+	TEST_IGNORE();
+#endif
+
+	create_ascii_set(1);
+	asciiStr[ASCII_LENGTH / 2] = '\0';
+	char holder[ASCII_LENGTH],
+		*rest = NULL;
+
+	/* Last element of ascii string is \127 and between 0 and 127 was written 0*/
+	memcpy(holder, strtok_r(asciiStr, "\127", &rest), ASCII_LENGTH);
```

Comment:
Warnings: 
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/116297248/d2112936-5e49-4801-b2d2-bff0997f8237)

This warning is misleading, it's caused by lack of `strtok_r `on Phoenix-RTOS, the code itself is fine :)